ui/src/components/admin/sermonNotes/SermonNotesEditorModal.js

```diff
@@ -0,0 +1,423 @@
+import React, { useEffect } from 'react';
+import {
+  Container,
+  FormControl,
+  Box,
+  FormLabel,
+  Input,
+  Grid,
+  Stack,
+  Button,
+  useToast,
+  Select,
+  FormErrorMessage,
+} from '@chakra-ui/react';
+import { useForm } from 'react-hook-form';
+import { useState } from 'react';
+import { customAxios as axios } from '../../helpers/customAxios';
+
+const SermonNotesEditorModal = (props) => {
+  const { editSermonNotesData, actionOnEditor, setIsEditorOpen } = props;
+  const { register, handleSubmit, setValue } = useForm();
+  const toast = useToast();
+
+  // Create a object
+  const sermonIdMap = {
+    'Sunday Celebration': 'sn',
+    Encounter: 'encounter',
+  };
+
+  const [sermondId, setSermonId] = useState('');
+  const [sermonNoteData, setSermonNoteData] = useState({
+    title: '',
+    subtitle: '',
+    speaker: '',
+    sermonSeries: '',
+    imageLink: '',
+    originalContent: '',
+    sermonLink: '',
+    serviceType: '',
+    date: '',
+    passage: '',
+  });
+
+  // This is the state that will hold the number of sermons for a particular date for edge case handling
+  const [numberOfSermons, setNumberOfSermons] = useState(0);
+
+  const editorSubmitButton = (actionOnEditor) => {
+    switch (actionOnEditor) {
+      default:
+        return 'Create Sermon Notes';
+      case 'edit':
+        return 'Update Sermon Notes';
+      case 'duplicate':
+        return 'Duplicate Sermon Notes';
+    }
+  };
+
+  const setSermonNotesData = (data) => {
+    if (data) setValue('title', data.title);
+    setValue('subtitle', data.subtitle);
+    setValue('speaker', data.speaker);
+    setValue('sermonSeries', data.sermonSeries);
+    setValue('date', data.date);
+    setValue('imageLink', data.imageLink);
+    setValue('originalContent', data.originalContent);
+    setValue('sermonLink', data.sermonLink);
+    setValue('serviceType', data.serviceType);
+    setValue('passage', data.passage);
+
+    setSermonNoteData({
+      title: data.title,
+      subtitle: data.subtitle,
+      speaker: data.speaker,
+      sermonSeries: data.sermonSeries,
+      imageLink: data.imageLink,
+      originalContent: data.originalContent,
+      sermonLink: data.sermonLink,
+      serviceType: data.serviceType,
+      date: data.date,
+      passage: data.passage,
+    });
+  };
```

Comment:
This is redundant since you already have the useState for `setSermonNoteData`. 
If you do this so you don't have to set the data separately everytime you wanna update the data, you can just do something like `setSermonNoteData(...data)` or  `setSermonNoteData({...data})`. I forgot the proper syntax lol. u can use this in things like the onSubmit